05/app.js

```diff
@@ -0,0 +1,65 @@
+const formEL = document.querySelector('form');
+const messagesUlEl = document.querySelector('.messages');
+
+formEL.addEventListener('submit', validateData);
+
+function validateData(e) {
+
+    e.preventDefault();
+    const errors = [];
+
+    const firstName = e.target.elements.firstName;
+    const lastName = e.target.elements.lastName;
+    const streetName = e.target.elements.street;
+    const houseNumber = e.target.elements.houseNumber;
+    const flatNumber = e.target.elements.flatNumber;
+    const zip = e.target.elements.zip;
+    const city = e.target.elements.city;
+    const voivodeship = e.target.elements.voivodeship;
+
+    if (firstName.value.length === 0) {
+        errors.push('Field Imię is required')
+    }
+
+    if (lastName.value.length === 0) {
+        errors.push('Field Nazwisko is required')
+    }
+
+    if (houseNumber.value.length === 0 || Number(houseNumber.value) === NaN) {
+        errors.push('House has to have a numeric value')
+    }
```

Comment:
Może zamiast mnóstwa if-ów dałoby się to inaczej napisać?
Tutaj podpowiedź: https://devmentor.pl/b/walidacja-formularza-w-javascript